### ISSUE TITLE
Configuration for boolean properties do not have common convention

### ISSUE THREAD

**Issue by kokosing** [prestosql/presto#6264]
```
kokosing@m16:~/presto$ git grep @Config | grep enabled | grep '\.enabled' | wc -l
26
kokosing@m16:~/presto$ git grep @Config | grep enabled | grep '\-enabled' | wc -l
25
```

The goal of this task is to make all such properties be coherent and intuitive as much as possible. We should use convention `feature.enabled`. For example `@Config("hive.non-managed-table-writes-enabled")` should become `@Config("hive.non-managed-table-writes.enabled")`

**Comment by losipiuk** [comment]
Old config version should still work. This can be achieved via `@LegacyConfig`

**Comment by findepi** [comment]
> For example `@Config("hive.non-managed-table-writes-enabled")` should become `@Config("hive.non-managed-table-writes.enabled")`

This example is concerning. This is just plain boolean flag. There is no "non managed table writes" feature, so this doesn't really fall into `feature.enabled` pattern for me.

**Comment by kokosing** [comment]
Yes this example is complex. To me it is a bit misnamed. By "feature" I mean certain functionality that could be enabled or disabled. 

How about `hive.mutable-non-managed-table.enabled`?
